src/unix.rs

```diff
@@ -330,17 +330,13 @@ pub fn lock(fd: FileDesc) -> Result<(), Error> {
 }
 
 /// Tries to lock a file but returns as soon as possible if already locked.
-pub fn try_lock(fd: FileDesc) -> Result<bool, Error> {
+pub fn try_lock(fd: FileDesc) -> Result<(), Error> {
```

Comment:
If you decide not adding the method, change here too:

```suggestion
pub fn try_lock(fd: FileDesc) -> Result<(), TryLockError> {
```